easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1-fosscuda-2017b-Python-2.7.14.eb

```diff
@@ -0,0 +1,98 @@
+easyblock = 'PythonPackage'
+
+name = 'PyTorch'
+version = '0.3.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://pytorch.org/'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
+PyTorch is a deep learning framework that puts Python first."""
+
+toolchain = {'name': 'fosscuda', 'version': '2017b'}
+
+pytorchdir = 'pytorch-%(version)s'
+extract_cmd_pattern = 'tar -C %s/%s --strip-components=1 -xf %%s'
+
+source_urls = [
+    'https://github.com/pytorch/pytorch/archive',
+    "https://github.com/facebookincubator/gloo/archive",
+    "https://github.com/google/googletest/archive",
+    "https://github.com/nanopb/nanopb/archive",
+    "https://github.com/pybind/pybind11/archive",
+    "https://github.com/wjakob/clang-cindex-python3/archive",
+]
+# PyTorch pulls in a bunch of submodules which don't have releases.
+# We download the submodule revisions from their repos.
+sources = [
+    'v%(version)s.tar.gz',  # PyTorch
+    {
+        'download_filename': 'cb002e4eb8d167c2c60fc3bdaae4e1844e0f9353.tar.gz',
+        'filename': 'gloo-20180105.tar.gz',
+        'extract_cmd': extract_cmd_pattern % (pytorchdir, 'torch/lib/gloo'),
+    },
+    {
+        'download_filename': 'ec44c6c1675c25b9827aacd08c02433cccde7780.tar.gz',
+        'filename': 'googletest-20160714.tar.gz',
+        'extract_cmd': extract_cmd_pattern % (pytorchdir, 'torch/lib/gloo/third-party/googletest'),
+    },
+    {
+        'download_filename': '14efb1a47a496652ab08b1ebcefb0ea24ae4a5e4.tar.gz',
+        'filename': 'nanopb-20170723.tar.gz',
+        'extract_cmd': extract_cmd_pattern % (pytorchdir, 'torch/lib/nanopb'),
+    },
+    {
+        'download_filename': '9f6a636e547fc70a02fa48436449aad67080698f.tar.gz',
+        'filename': 'pybind11-20170821.tar.gz',
+        'extract_cmd': extract_cmd_pattern % (pytorchdir, 'torch/lib/pybind11'),
+    },
+    {
+        'download_filename': '6a00cbc4a9b8e68b71caf7f774b3f9c753ae84d5.tar.gz',
+        'filename': 'clang-cindex-python3-20170330.tar.gz',
+        'extract_cmd': extract_cmd_pattern % (pytorchdir, 'torch/lib/pybind11/tools/clang'),
+    },
+]
+patches = [
+    '%(name)s-%(version)s-cmake.patch',
+    '%(name)s-%(version)s_skip_MPI_tests.patch',
+]
```

Comment:
Any reason you left out PyTorch-0.3.1_fix_findAVX.patch ?

Reply:
From the existing easyconfig files I inferred that that patch is required only for builds done with the Intel compiler. Do you advise otherwise?

Reply:
The patch for 0.3.1 says:
"original AVX and AVX2 tests were passed by icc compiler, the new code works well for both gcc and icc"
I interpret that to mean that it is needed for GCC based builds.

But in either case, it's better to keep a synchronized set of patches as much as possible between toolchains.